### PR TITLE
[4.3] phpmd: Fixing missing variable declarations

### PR DESCRIPTION
**Pull request by Hackwar**
### Summary of Changes
I ran `phpmd` on the `libraries/src` folder, only checking for undefined variables and these are the results. 

### Testing Instructions
This needs code review and can not be tested by a normal user.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: libraries/src/Application/CMSApplication.php [hunk 1]
```diff
@@ -922,8 +922,10 @@ public function logout($userid = null, $options = [])
         $user = Factory::getUser($userid);
 
         // Build the credentials array.
-        $parameters['username'] = $user->get('username');
-        $parameters['id'] = $user->get('id');
+        $parameters = [
+            'username' => $user->get('username'),
+            'id' => $user->get('id')
```
Comment (sandewt) [inline libraries/src/Application/CMSApplication.php]: Missing a comma:

```
        $parameters = [
            'username' => $user->get('username'),
            'id' => $user->get('id')
        ];
```
I often see a comma at the end. This also applies to other changes.
```
        $parameters = [
            'username' => $user->get('username'),
            'id' => $user->get('id'),
        ];
```

Comment (laoneo) [reply libraries/src/Application/CMSApplication.php]: @sandewt thanks for the hint. In #39745 we are properly checking for trailing comma, so we can ignore it for now.

**Comment by Hackwar** [comment]
The trailing commas should be solved now.

**Comment by laoneo** [comment]
Thanks!

File: libraries/src/Application/CMSApplication.php (hunk 1 above)
Comment (sandewt) [reply libraries/src/Application/CMSApplication.php]: @laoneo 

What I find confusing is that @nikosdion  doesn't apply a trailing comma either in the folowing examples:

see: https://github.com/joomla/joomla-cms/issues/39750#issuecomment-1408876022

Maybe he has a reason for that ?

Comment (laoneo) [reply libraries/src/Application/CMSApplication.php]: I do that also not for my extensions and honestly I don't like a trailing comma on arrays. But this is a rule for our code style, so the code base should be consistent in the CMS.

Comment (nikosdion) [reply libraries/src/Application/CMSApplication.php]: I very quickly wrote code directly on GitHub's comment editor. Remember that the intent of the comment was to show you when you don't need to use prepared statements, not showcase the Joomla code style. You will see that even the indentation does not match the Joomla code style.

Trailing commas in lists (arrays) are, generally speaking, a good thing. It reduces the likelihood of a developer screwing up if they move lines around, or by simply adding another element at the end of the list.

Comment (sandewt) [reply libraries/src/Application/CMSApplication.php]: Thanks for the info, I'm clear now.

I also see it in the manual:  https://developer.joomla.org/coding-standards/php-code.html

**Pull request merged**